src/domain/community/user/pages/UserMembershipPage.tsx

```diff
@@ -8,36 +8,84 @@ import UserSettingsLayout from '../../../platform/admin/user/layout/UserSettings
 import { useUserMetadata } from '../hooks/useUserMetadata';
 import GridProvider from '../../../../core/ui/grid/GridProvider';
 import SectionSpacer from '../../../shared/components/Section/SectionSpacer';
-import useUserContributions from '../userContributions/useUserContributions';
 import { SpaceHostedItem } from '../../../journey/utils/SpaceHostedItem';
+import { CommunityContributorType, SpaceType } from '../../../../core/apollo/generated/graphql-schema';
+import {
+  useUserContributionsQuery,
+  useUserPendingMembershipsQuery,
+} from '../../../../core/apollo/generated/apollo-hooks';
+import { JourneyLevel } from '../../../../main/routing/resolvers/RouteResolver';
 
 export interface UserMembershipPageProps {}
 
 const UserMembershipPage: FC<UserMembershipPageProps> = () => {
   const { t } = useTranslation();
   const { userNameId = '' } = useUrlParams();
-  const { user: userMetadata, loading } = useUserMetadata(userNameId);
+  const { user: userMetadata } = useUserMetadata(userNameId);
 
-  const contributions = useUserContributions(userNameId);
+  const { data, loading, refetch } = useUserContributionsQuery({
+    variables: {
+      userId: userMetadata?.user.id!,
+    },
+    skip: !userMetadata?.user.id,
+  });
 
-  const applications = useMemo<SpaceHostedItem[] | undefined>(
-    () =>
-      userMetadata?.pendingApplications.map(application => ({
+  const memberships = useMemo(() => {
+    if (!data?.rolesUser.spaces) {
+      return [];
+    }
+
+    return data.rolesUser.spaces.reduce((acc, space) => {
+      const currentSpace = {
+        spaceID: space.id,
+        id: space.id,
+        spaceLevel: 0 as JourneyLevel,
+        contributorId: userMetadata?.user.id!,
+        contributorType: CommunityContributorType.User,
+      };
+      acc.push(currentSpace);
+
+      const subspaces = space.subspaces.map(subspace => ({
+        id: subspace.id,
+        spaceID: subspace.id,
+        spaceLevel:
+          subspace.type === SpaceType.Challenge
```

Comment:
Can we query directly the level instead of the type and skip this obsolete logic?